code/modules/power/lighting.dm

```diff
@@ -745,11 +818,11 @@
 
 /obj/item/light/proc/shatter()
 	if(status == LIGHT_OK || status == LIGHT_BURNED)
-		src.visible_message("<span class='warning'>[name] shatters.</span>","<span class='warning'>You hear a small glass object shatter.</span>")
+		visible_message("<span class='warning'>[src] shatters.</span>", "<span class='warning'>You hear a small glass object shatter.</span>")
```

Comment:
it will now show the "You hear a glass object shatter" message to the light. I think you meant to make that message be the `blind_message`. You can do this by adding a null as 2nd param

Reply:
The `self_message` parameter is only on `/mob/visible_message`, whereas this is just `/atom/proc/visible_message`.

Reply:
Yep you're right! Overlooked that version of the proc